pyquil/_parser/PyQuilListener.py

```diff
@@ -148,28 +148,49 @@ def exitDefCircuit(self, ctx: QuilParser.DefCircuitContext):
 
     def exitGate(self, ctx: QuilParser.GateContext):
         gate_name = ctx.name().getText()
+        modifiers = [mod.getText() for mod in ctx.modifier()]
         params = list(map(_param, ctx.param()))
         qubits = list(map(_qubit, ctx.qubit()))
-        # The Gate.controlled() method *prepends* the CONTROLLED modifier to the gate. But the
-        # parser works from the outside-in. Therefore the controlled qubits would be in reverse
-        # order. We reverse here to fix that.
-        modifiers = [mod.getText() for mod in ctx.modifier()][::-1]
-        control_qubits = qubits[0:len(list(filter(lambda str: str == "CONTROLLED", modifiers)))][::-1]
-        target_qubits = qubits[len(control_qubits):]
+
+        # The parsed string 'DAGGER CONTROLLED X 0 1' gives
+        #   modifiers ['DAGGER', 'CONTROLLED']
+        #   qubits    ['0', '1']
+        #
+        # We will build such gates up from "the inside out", e.g.
```

Comment:
Is "inside out" correct? Wouldn't that mean starting with `X 0`?

Reply:
I updated the doc a bit to clarify.